File: test-paddle-ocr.py
from paddleocr import PaddleOCR
import sys
import json
from pathlib import Path


def main():
    if len(sys.argv) < 2:
        print("Usage:")
        print('python test-paddle-ocr.py "path/to/image.png"')
        sys.exit(1)

    image_path = sys.argv[1]

    if not Path(image_path).exists():
        print(f"Image not found: {image_path}")
        sys.exit(1)

    print("Initializing PaddleOCR...")

    # English OCR, CPU
    ocr = PaddleOCR(
        lang="en",
        device="cpu"
    )

    print("Running OCR...")
    print(f"Input: {image_path}")
    print()

    result = ocr.predict(image_path)

    print("========== RAW RESULT ==========")
    print(result)

    print()
    print("========== OCR TEXT ==========")

    found = False

    for res in result:
        data = res.json

        if callable(data):
            data = data()

        # PaddleOCR 3.x result structure
        rec_texts = data.get("res", {}).get("rec_texts", [])
        rec_scores = data.get("res", {}).get("rec_scores", [])

        for text, score in zip(rec_texts, rec_scores):
            found = True
            print(
                f"Text: {text} | "
                f"Confidence: {float(score):.4f}"
            )

    if not found:
        print("No text detected.")


if __name__ == "__main__":
    main()

File: image-pipeline/test-paddle-ocr.js
const {
  extractPlateText
} = require('./src/services/paddleOCR');

async function test() {
  try {
    const imagePath = process.argv[2];

    if (!imagePath) {
      console.log(
        'Usage: node test-paddle-ocr.js "path/to/image.png"'
      );
      process.exit(1);
    }

    console.log('PaddleOCR Node Test');
    console.log('');
    console.log('Input:', imagePath);
    console.log('');

    const result =
      await extractPlateText(imagePath);

    console.log('✓ OCR complete');
    console.log('');
    console.log('Plate text:', result.text);
    console.log('Raw text:', result.rawText);

    console.log('');
    console.log('Lines:');

    for (let i = 0; i < result.lines.length; i++) {
      const line = result.lines[i];

      console.log(
        `  ${i + 1}. ${line.text} ` +
        `(confidence: ${line.confidence})`
      );
    }

  } catch (err) {
    console.error('❌ Error:', err.message);
    process.exit(1);
  }
}

test();